packages/amazon/src/serverGroup/configure/serverGroupCommandBuilder.service.js

```diff
@@ -359,49 +378,127 @@ angular
               const shouldAutoEnableIPv6 =
                 asgSettings && asgSettings.enableIPv6 && asgSettings.setIPv6InTest && isTestEnv;
 
+              const secGroups = launchTemplateData.networkInterfaces
+                ? (launchTemplateData.networkInterfaces.find((ni) => ni.deviceIndex === 0) || {}).groups
+                : launchTemplateData.securityGroups.length && launchTemplateData.securityGroups;
+
               angular.extend(command, {
                 instanceType: launchTemplateData.instanceType,
                 iamRole: launchTemplateData.iamInstanceProfile.name,
                 keyPair: launchTemplateData.keyName,
                 associateIPv6Address: shouldAutoEnableIPv6 || Boolean(ipv6AddressCount),
                 ramdiskId: launchTemplateData.ramdiskId,
-                instanceMonitoring: launchTemplateData.monitoring.enabled,
+                instanceMonitoring: launchTemplateData.monitoring && launchTemplateData.monitoring.enabled,
                 ebsOptimized: launchTemplateData.ebsOptimized,
-                spotPrice: maxPrice || undefined,
+                spotPrice: spotMaxPrice || undefined,
                 requireIMDSv2: Boolean(
                   launchTemplateData.metadataOptions && launchTemplateData.metadataOptions.httpsTokens === 'required',
                 ),
                 unlimitedCpuCredits: launchTemplateData.creditSpecification
                   ? launchTemplateData.creditSpecification.cpuCredits === 'unlimited'
                   : undefined,
+                securityGroups: secGroups,
               });
 
               command.viewState.imageId = launchTemplateData.imageId;
             }
 
-            if (mode === 'clone' && serverGroup.image && serverGroup.image.name) {
-              command.amiName = serverGroup.image.name;
-            }
+            if (serverGroup.mixedInstancesPolicy) {
+              const mip = serverGroup.mixedInstancesPolicy;
+              const { launchTemplateData } = mip.launchTemplates[0]; // note: single launch template case is currently the only supported case for mixed instances policy
 
-            if (serverGroup.launchConfig && serverGroup.launchConfig.securityGroups.length) {
-              command.securityGroups = serverGroup.launchConfig.securityGroups;
-            }
+              const { ipv6AddressCount } =
+                launchTemplateData.networkInterfaces &&
+                launchTemplateData.networkInterfaces.length &&
+                launchTemplateData.networkInterfaces[0];
```

Comment:
This snippet also exists under the [launchTemplate code path ](https://github.com/spinnaker/deck/blob/master/packages/amazon/src/serverGroup/configure/serverGroupCommandBuilder.service.js#L352)- 
and relies on the fact that Clouddriver always creates a network interface with a launch template - https://github.com/spinnaker/clouddriver/blob/master/clouddriver-aws/src/main/java/com/netflix/spinnaker/clouddriver/aws/services/LaunchTemplateService.java#L456 

You are right - if the launch template creation is changed in Clouddriver, this could blow up. I will put a safety check in, just in case.